src/as_execution/abi.rs

```diff
@@ -1051,6 +1051,14 @@ pub fn assembly_script_function_exists(
     Ok(function_exists(&mut ctx, address, function)? as i32)
 }
 
+/// Return current chain id
+#[named]
+pub(crate) fn assembly_script_chain_id(mut ctx: FunctionEnvMut<ASEnv>) -> ABIResult<i64> {
+    let env = get_env(&ctx)?;
+    sub_remaining_gas_abi(&env, &mut ctx, function_name!())?;
+    Ok(env.get_interface().chain_id()? as i64)
```

Comment:
Don't really remember, but below it's a u64. Is it mapped correctly with i64?

Reply:
Fixed